src/torrent.cpp

```diff
@@ -5790,8 +5790,8 @@ namespace libtorrent
 			return;
 		}
 
-		bool const is_ip = is_ip_address(hostname.c_str());
-		if (is_ip) a.address(address::from_string(hostname.c_str(), ec));
+		bool const is_ip = is_ip_address(hostname);
+		if (is_ip) a.address(address::from_string(hostname, ec));
```

Comment:
heh, this code is a bit weird (unrelated to your patch). ``is_ip_address()`` just calls ``from_string`` and checks the error, it seems redundant to just call it twice here